go/common/grpc/proxy/proxy.go

```diff
@@ -0,0 +1,207 @@
+// Package proxy implements service agnostic gRPC reverse proxy.
+//
+// This package is somewhat inspired by https://github.com/mwitkow/grpc-proxy.
+package proxy
+
+import (
+	"context"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/oasislabs/oasis-core/go/common/cbor"
+	policy "github.com/oasislabs/oasis-core/go/common/grpc/policy/api"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+)
+
+// Handler returns a grpc StreamHandler than can be used
+// to proxy requests to provided client.
+// XXX: potentially the connection should be established in this package,
```

Comment:
Plan to file issues around these?

Reply:
yeah i'll file a single issue about potential proxy benchmarks and possible improvements, since i feel like for most of these some benchmarks should be first done (e.g. compare perf of accessing a storage node directly or via a sentry)